src/client/cosmos.ts

```diff
@@ -655,6 +636,10 @@ export default class CosmosDbClient {
       (err && retryableNetworkingErrors.includes(err.code))
     );
   }
+
+  private static _formatEndpoint(accountName: string): string {
+    return `https://${accountName}.documents.azure.com:443/`;
```

Comment:
ask for the endpoint, not the name. the `documents.azure.com` suffix only applies to public cloud so it's important that we allow people to set the endpoint that is relevant to them